core/src/main/java/org/hibernate/ogm/dialect/storedprocedure/spi/StoredProcedureAwareGridDialect.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.dialect.storedprocedure.spi;
+
+import java.util.Map;
+
+import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
+import org.hibernate.ogm.dialect.spi.GridDialect;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+
+/**
+ * A facet for {@link GridDialect} implementations which support the execution of stored procedures.
+ *
+ * Cases of stored procedures are :
+ * <ou>
+ *     <li>procedure without any input or output parameters</li>
+ *	   <li>function with many input parameters and one returned value</li>
+ * </ou>
+ *
+ *
+ * @see <a href="https://stackoverflow.com/questions/32480060/call-mongodb-function-from-java">Example for MongoDB</a>
+ * @see <a href="http://orientdb.com/docs/2.0/orientdb.wiki/Functions.html">Example for OrientDB</a>
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public interface StoredProcedureAwareGridDialect extends GridDialect {
+
+	/**
+	 * Is data storage supports parameter position by name (true) or by index (false)
+	 *
+	 * @return
+	 */
+	boolean supportsNamedPosition();
+
+
+	/**
+	 * Returns the result of a stored procedure executed on the backend.
+	 * Tne method uses for storages that supports name position
+	 *
+	 * @param storedProcedureName name of stored procedure.
+	 * @param params - array with values
+	 * @param tupleContext the tuple context
+	 *
+	 * @return an {@link ClosableIterator} with the result of the query
+	 */
+
+	Object callStoredProcedure( String storedProcedureName, Object[] params, TupleContext tupleContext);
```

Comment:
In this way, stored procedure have to return a collection of Entity.  But ... What will we do with case then stored procedure returns simple type?  For example, Integer

Reply:
A tuple is not necessary an entity, it can contain anything.
 
You can check the implementations of `QueryableGridDialect#executeBackendQuery` to see what we already do. You will find that what we return is an iterator of map containing many different values or an iterator of the underlying object that represents the entity in the db, for example nodes in Neo4j

Reply:
If you need to return only one value it will be a tuple with only one column.

Reply:
ok. I will investigate